column_index.go

```diff
@@ -145,8 +143,8 @@ func (i byteArrayColumnIndex) NullCount(int) int64 { return 0 }
 func (i byteArrayColumnIndex) NullPage(int) bool   { return false }
 func (i byteArrayColumnIndex) MinValue(int) Value  { return makeValueBytes(ByteArray, i.page.min()) }
 func (i byteArrayColumnIndex) MaxValue(int) Value  { return makeValueBytes(ByteArray, i.page.max()) }
-func (i byteArrayColumnIndex) IsAscending() bool   { return bytes.Compare(i.page.bounds()) < 0 }
-func (i byteArrayColumnIndex) IsDescending() bool  { return bytes.Compare(i.page.bounds()) > 0 }
+func (i byteArrayColumnIndex) IsAscending() bool   { return true }
```

Comment:
Is the reasoning that it is safe for this to be in ascending order because there is only one page?

I'm also wondering whether it would be the right decision to make, maybe we should assume that we don't know what the order is if there is a single page?

Reply:
That's correct. We basically have to pick one and my take is that the choice has already been made for us by `fileColumnIndex`. It uses `internal/bits/order` to determine order which [returns ascending for a single-length slice](https://github.com/segmentio/parquet-go/blob/9409f0293efe3f41863e20a01a07d076bfe6950e/internal/bits/order.go#L56-L57). It has a vectorized assembly implementation which is harder to change than this code here.

Reply:
The `bits.OrderOf*` functions are implemented in Go right? only the internal `bits.orderOf*` functions have assembly-optimized version, or am I reading this wrong?

Reply:
🤦 You're right. We could change the implementation of the `len == 1` case globally without touching the optimized code.

I'll update the code to treat single-item slices as unordered.

Reply:
Updated in db705aa